sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HostPolicy.java

```diff
@@ -39,7 +39,8 @@ public HostPolicy(String host) {
             context.getHttpRequest().setUrl(urlBuilder.setHost(host).toURL());
             result = next.process();
         } catch (MalformedURLException e) {
-            result = Mono.error(e);
+            result = Mono.error(new RuntimeException(String.format("Host URL %s is invalid ",
+                urlBuilder.setHost(host)), e));
```

Comment:
Are you also invoking a method in here to set the host? Is this a desired side-effect?

Reply:
Yeah, this looks really odd. We shouldn't expect side effects from an exception.

Reply:
Yes. Good catch. 
The right logging is String.format("Host URL %s is invalid ", host)